### PR TITLE
Upload images in the comments box

### LINKED ISSUE #2692: Reuse the image uploader for comments as well, but using CDN2
### Is your request related to a problem? Please describe.

Always, I have to go to cubeupload to upload an image to show people in the comments

### Describe the solution you'd like

Adding the image uploader on comment boxes. Note that assets automatically downloads the image, but CDN2 doesnt.

### Describe alternatives you've considered

Using a bookmarklet, but i cannot copy images from the clipboard.

### PR DESCRIPTION
**Pull request by FunctionalMetatable**
Resolves #2692

### Changes

When you drag/paste an image in the comments section, it uploads the image to CDN2.

### Reason for changes

People have to go to cubeupload/use assets (which is bad) to upload images to comments

### Tests

Tested locally

Known bugs:
~~When you paste an image in your own profile page, your profile picture gets changed. Not sure how to fix this~~ fixed

cc @jeffalo

### CODE REVIEW

**Comment by RedGuy12** [comment]
Use assets not cdn2. To make it not download the image, remove the `/get_image/.%2E` part. That’s just a image whitelist hack only needed in the forums.

**Comment by FunctionalMetatable** [comment]
> Use assets not cdn2. To make it not download the image, remove the `/get_image/.%2E` part. That’s just a image whitelist hack only needed in the forums.

Still downloads it even if I remove the get_image part. Cdn2 doesn't download it.

**Comment by RedGuy12** [comment]
oh ok nvm then

**Comment by jeffalo** [comment]
looks neat. clean up the requests to `https://scratch.mit.edu/site-api/projects/all/${projectId}/` all you need is the thing you're changing. everything else isn't needed. oh and try making it JSON.stringify()

and set a project status to delbyusr to delete it without putting it in the trash

File: addons/image-uploader/addon.json [hunk 1]
```diff
@@ -1,15 +1,23 @@
 {
-  "name": "Forums image uploader",
-  "description": "Integrates image uploading into the forums.",
+  "name": "Image uploader",
```
Comment (Shluffy) [inline addons/image-uploader/addon.json:2]: maybe have it be Comment & forums image uploader?

Comment (FunctionalMetatable) [reply addons/image-uploader/addon.json:2]: Too long

Comment (Shluffy) [reply addons/image-uploader/addon.json:2]: that would change it to a phrase (#2764), but that would only be 32 chars, or 11 more. There are definitley longer ones. But #2764 might mean that it shouldn't be done?

**Comment by WorldLanguages** [comment]
I'm not sure if creating a new project for every new uploaded image is reasonable 🤔

**Comment by FunctionalMetatable** [comment]
> I'm not sure if creating a new project for every new uploaded image is reasonable 🤔

...but it gets deleted without putting it in the trash

**Comment by mxmou** [comment]
> I'm not sure if creating a new project for every new uploaded image is reasonable 🤔

It would also be possible to use a tool like https://gosoccerboy5.github.io/view-images/ to create links to assets images that don't auto-download.

**Comment by WorldLanguages** [comment]
> It would also be possible to use a tool like https://gosoccerboy5.github.io/view-images/ to create links to assets images that don't auto-download.

But then the link wouldn't be clickable on vanilla Scratch.

**Comment by WorldLanguages** [comment]
Even tho forums image uploader did this in the past, I don't find it reasonable to create a new project each time a new image is uploaded for use in comments

`content-disposition: attachment` header in assets server is unfortunate, and there's probably no way around this, unless the Scratch Wiki adds an endpoint that inlines an image from the Scratch asset server, or something like that.

**Pull request closed without merging**